core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -154,15 +155,17 @@ private[spark] class HighlyCompressedMapStatus private (
 
   override def writeExternal(out: ObjectOutput): Unit = Utils.tryOrIOException {
     loc.writeExternal(out)
-    emptyBlocks.writeExternal(out)
+    markedBlocks.writeExternal(out)
```

Comment:
In this example, calling ``runOptimize`` should not change anything. However, if you try something like this:

```
    val r = new RoaringBitmap
    for (i <- 1 to 200000) {
      if(i % 200 != 0)
        r.add(i)
    }
    val size1 = r.getSizeInBytes
    r.runOptimize()
    r.trim()
    val size2 = r.getSizeInBytes
```

... you should get close to a 10x saving.